app/helpers/puppetclasses_and_environments_helper.rb

```diff
@@ -17,7 +17,7 @@ def import_proxy_select(hash)
 
   def import_proxy_links(hash, classes = nil)
     SmartProxy.with_features("Puppet").map do |proxy|
-      display_link_if_authorized(_("Import from %s") % proxy.name, hash.merge(:proxy => proxy), :class=>classes)
+      display_link_if_authorized(_("Import from %s") % proxy.name, hash.merge(:proxy => proxy), {:class=>classes, :"data-no-turbolink" => true})
```

Comment:
IMHO that would be more complex then a simple ajax call to start with, we already added a caching layer on the proxy, and there are two longish operations in the background
1. ask the proxy to scan for puppet manifests file, the proxy already caches the results, but we could consider adding a new api for all environments vs per environment api.
2. compare the current state in the db (env -> classes -> parameters) to the new state. also takes time.
the first action is what slows down the initial page load, and the second action is what slows down after submitting... both can be very long.